nixos/modules/services/security/keylime.nix

```
@@ -0,0 +1,103 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.keylime;
+in
+{
+  options = {
+    services.keylime = {
+      enable = mkOption {
+        default = false;
+        type = types.bool;
+        description = lib.mdDoc "Whether to enable keylime services.";
+      };
+
+      registrar = {
+        enable = mkOption {
+          default = false;
+          type = types.bool;
+          description = lib.mdDoc "Whether to enable keylime-registrar service.";
+        };
+      };
+
+      verifier = {
+        enable = mkOption {
+          default = false;
+          type = types.bool;
+          description = lib.mdDoc "Whether to enable keylime-verifier service.";
+        };
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = "keylime";
+        description = lib.mdDoc "User which runs keylime.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "keylime";
+        description = lib.mdDoc "Group which runs keylime.";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable (mkMerge [
+    {
+      environment.etc."keylime/tenant.conf".source = "${pkgs.keylime}/etc/config/tenant.conf";
+
+      users.users.keylime = {
+        description = "Keylime user";
+        extraGroups = [ "${cfg.group}" ];
+        isSystemUser = true;
+      };
+
+      users.groups.${cfg.group} = {};
+    }
+
+    (mkIf cfg.registrar.enable {
+      environment.etc."keylime/registrar.conf".source = "${pkgs.keylime}/etc/config/registrar.conf";
+      environment.etc."keylime/logging.conf".source = "${pkgs.keylime}/etc/config/logging.conf";
+
+      systemd.services."keylime-registrar" = {
+        description = "Keylime registrar service";
+        after = [ "network.target" ];
+        wantedBy = [ "default.target" ];
+        serviceConfig = {
+          User = "${cfg.user}";
+          Group = "${cfg.group}";
```

Unnecessary string interpolations

```suggestion
          User = cfg.user;
          Group = cfg.group;
```